R/facet-.r

```diff
@@ -311,13 +311,16 @@ as_facets_list <- function(x) {
     x <- lapply(x, as_facets)
   }
 
-  if (sum(vapply(x, length, integer(1))) == 0L) {
-    stop("Must specify at least one variable to facet by", call. = FALSE)
-  }
-
   x
 }
 
+# Flatten a list of quosures objects to a quosures object, and compact it
+compact_facets <- function(x) {
+  x <- rlang::flatten_if(x, rlang::is_list)
+  null <- vapply(x, rlang::quo_is_null, logical(1))
+  rlang::as_quosures(x[!null])
```

Comment:
I think this should be `new_quosures()`. The casting form converts formulas to quosures etc, and at this point we should have standardised the inputs already.

Reply:
Makes sense. Thanks!